### ISSUE TITLE
ASP.Net Add-in and Outlook

### ISSUE THREAD

**Issue by elthombre** [OfficeDev/Office-Add-in-ASPNET-SSO#46]
Hi,

I wish there was a good Outlook add-in example written in ASP.Net. 

I am in a situation wherein it was first determined that a pre-existing, large corporate website would be used for authentication against an add-in created in Angular. There are many security config details with this ASP.Net solution. While it appears the popup DialogAsync may have solved the authentication, which relies on HTTPOnly cookies, now it has been determined that the website's UI will be tweaked to fit into the taskpane.

In other words: use the Angular add-in to do nothing more than attempt to serve up ASP.Net pages.

Does this make sense?

To me, it would appear, an add-in scaffolding should be created in ASP.Net and then the components needed be reused from the website. Then, the solution is all within a single application which also eliminates the issues associated with security in relation to the much larger, web site.

Please advise.

Thanks

**Comment by Rick-Kirkham** [comment]
I don't have a clear picture of what you are asking, but for what it is worth: 

1. Angular and ASP.Net MVC are both MVWhatever frameworks. Mixing them together seems confused and error prone. 
2.  It should be possible to convert this sample to and Outlook sample. The main change would be swapping out the manifest for an Outlook manifest. For example, the following two samples are virtually identical, except that the first is for Excel/Word/PowerPoint and the second is for Outlook. 

https://github.com/OfficeDev/PnP-OfficeAddins/tree/master/Samples/auth/Office-Add-in-Microsoft-Graph-ASPNET
https://github.com/OfficeDev/PnP-OfficeAddins/tree/master/Samples/auth/Outlook-Add-in-Microsoft-Graph-ASPNET

If you do a diff on the files in these two samples, you'll see what was changed to turn the first one into the second.

**Comment by elthombre** [comment]
Thanks for this. I wish there were a place in this OfficeDev group to discuss architecture. In the meantime, I was also wondering:

1. Is there a way to address this OfficeDev group as a whole? From what I can see, it is on a repository-to-repository basis.

2. Is VSTO being deprecated? I ask because I am told by some at my job that it is. However, I've seen nothing from Microsoft stating such and, with each Office release, the object model is usually extended.

**Comment by Rick-Kirkham** [comment]
An architecture issue is probably best raised as a question on Stack Overflow. Be sure to tag the question with `office-js`.

VSTO is not being deprecated.

**Comment by elthombre** [comment]
Hi Rick, 

Is there a way to update the web.config of an Outlook for web add-in to prevent the "... cannot be displayed in a frame" error?

Thanks

**Comment by Rick-Kirkham** [comment]
This is really a distinct question and its not really about this sample per se. I recommend that you ask this question on Stack Overflow and tag it `office-js`. I'll close this issue now, but you can comment here again if you think it was closed in error.
